server/src/metadata.rs

```diff
@@ -0,0 +1,1310 @@
+        let database_path = metadata_manager
+            .get_data_folder_path()
+            .join(METADATA_SQLITE_NAME);
+        let connection = Connection::open(database_path).unwrap();
+
+        let mut statement = connection
+            .prepare("SELECT * FROM model_table_tags")
+            .unwrap();
+        let mut rows = statement.query(()).unwrap();
+
+        assert_eq!(
+            rows.next()
+                .unwrap()
+                .unwrap()
+                .get::<usize, String>(1)
+                .unwrap(),
+            "tag1"
+        );
+    }
+
+    #[test]
+    fn test_get_existing_tag_hash() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let mut metadata_manager = test_util::get_test_metadata_manager(temp_dir.path());
+
+        let model_table_metadata = test_util::get_model_table_metadata();
+        metadata_manager
+            .save_model_table_metadata(&model_table_metadata, vec![])
+            .unwrap();
+
+        metadata_manager
+            .get_or_compute_tag_hash(&model_table_metadata, &vec!["tag1".to_owned()])
+            .unwrap();
+        assert_eq!(metadata_manager.tag_value_hashes.keys().len(), 1);
+
+        // When getting the same tag hash again, it should just be retrieved from the cache.
+        let result = metadata_manager
+            .get_or_compute_tag_hash(&model_table_metadata, &vec!["tag1".to_owned()]);
+
+        assert!(result.is_ok());
+        assert_eq!(metadata_manager.tag_value_hashes.keys().len(), 1);
+    }
+
+    #[test]
+    fn test_normalize_table_name_lowercase_no_effect() {
+        assert_eq!(
+            "table_name",
+            MetadataManager::normalize_table_name("table_name")
+        );
+    }
+
+    #[test]
+    fn test_normalize_table_name_uppercase() {
+        assert_eq!(
+            "table_name",
+            MetadataManager::normalize_table_name("TABLE_NAME")
+        );
+    }
+
+    #[test]
+    fn test_save_table_metadata() {
+        // Save a table to the metadata database.
+        let temp_dir = tempfile::tempdir().unwrap();
+        let metadata_manager = test_util::get_test_metadata_manager(temp_dir.path());
+
+        let table_name = "table_name";
+        metadata_manager.save_table_metadata(table_name).unwrap();
+
+        // Retrieve the table from the metadata database.
+        let connection =
+            Connection::open(metadata_manager.metadata_database_path.to_str().unwrap()).unwrap();
+        let mut select_statement = connection
+            .prepare("SELECT table_name FROM table_metadata")
+            .unwrap();
+        let mut rows = select_statement.query(()).unwrap();
+
+        let row = rows.next().unwrap().unwrap();
+        let retrieved_table_name = row.get::<usize, String>(0).unwrap();
+        assert_eq!(table_name, retrieved_table_name);
+
+        assert!(rows.next().unwrap().is_none());
+    }
+
+    #[test]
+    fn test_register_tables() {
```

Comment:
Is something missing in this test? it does not seem like anything is asserted.

Reply:
The test succeeds if none of the `unwrap()`s fail, so no `assert!()`s are needed. A comment have been added to document this.